pkgs/development/python-modules/hypothesis.nix

```diff
@@ -21,7 +21,7 @@ buildPythonPackage rec {
     sha256 = "1s911pd3y9hvk0hq2fr6i68dqv1ciagryhgp13wgyfqh8hz8j6zv";
   };
 
-  checkInputs = stdenv.lib.optionals doCheck [ pytest flake8 flaky ];
+  checkInputs = stdenv.lib.optionals doCheck [ pytest pytest_xdist flake8 flaky ];
```

Comment:
`pytest_xdist` is an optional dependency (although I guess `flake8` and `flaky` are as well). Did you check evaluation? Its always tricky with `pytest` and `hypothesis` regarding infinite recursion